core/sdk/src/service/service_client.rs

```diff
@@ -54,35 +58,72 @@ pub struct ServiceClient {
     pub service_id: Option<ExtnId>,
     pub outbound_extn_rx: Arc<RwLock<Option<mpsc::Receiver<ApiMessage>>>>,
     pub outbound_service_rx: Arc<RwLock<Option<mpsc::Receiver<ServiceMessage>>>>,
+    extn_manifest: ExtnManifest,
```

Comment:
I'm not certain it's a good idea for ServiceClient to store the entire `extn_manifest` and `device_manifest`. Ideally, each service should set these <key, value> pairs as needed,  and ServiceClient should provide a generic hash map to hold these values.